packages/bpk-component-chip-group/src/BpkChipGroup.tsx

```diff
@@ -70,131 +70,172 @@ export type ChipItem = {
   hidden?: boolean;
 } & SingleSelectChipItem;
 
+export type InternalProps = {
+  chipGroupClassNames: string
+};
+
 export type CommonProps = {
   label?: string;
   ariaLabel?: string;
-  type?: ChipGroupType;
   chipStyle?: ChipStyleType;
-  // only required when type = rail
-  leadingNudgerLabel?: string;
-  trailingNudgerLabel?: string;
+  chips: ChipItem[];
+  ariaMultiselectable?: boolean;
 };
 
-export type ChipGroupProps = {
-  chips: ChipItem[];
+export type RailChipGroupProps = {
   stickyChip?: ChipItem;
-  ariaMultiselectable?: boolean;
+  leadingNudgerLabel?: string;
+  trailingNudgerLabel?: string;
+} & CommonProps;
+
+export type WrapChipGroupProps = {
 } & CommonProps;
 
+export type ChipGroupProps = (RailChipGroupProps & { type: typeof CHIP_GROUP_TYPES.rail } | WrapChipGroupProps & { type: typeof CHIP_GROUP_TYPES.wrap });
 
-const BpkChipGroup = ({
+const Chip = ({ ariaMultiselectable, chipIndex, chipItem, chipStyle }: { chipIndex: number, chipItem: ChipItem, chipStyle: ChipStyleType, ariaMultiselectable: boolean }, index: number) => {
```

Comment:
Extracted from the inline func to build a chip

Reply:
Can we remove the `index` param now we have `chipIndex`?